src/averageAges.js

```diff
@@ -55,10 +82,35 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const validPeople = people.filter(person => {
+    if (onlyWithSon) {
+      return person.sex === 'm' && person.mother !== undefined;
```

Comment:
You should not return 'undefined'. Instead, you should return a boolean value. You can use !! operator to convert undefined to boolean false.